evennia/contrib/turnbattle/tb_magic.py

```diff
@@ -0,0 +1,1293 @@
+                    target_candidates.append(thing)
+                    
+        # Now, match each entry in spell_targets to an object in the search candidates
+        matched_targets = []
+        for target in spell_targets:
+            match = caller.search(target, candidates=target_candidates)
+            matched_targets.append(match)
+        spell_targets = matched_targets
+                    
+        # If no target is given and the spell's target is 'self', set target to self
+        if len(spell_targets) == 0 and spelldata["target"] == "self":
+            spell_targets = [caller]
+                    
+        # Give error message if trying to cast an "other" target spell on yourself
+        if spelldata["target"] in ["other", "otherchar"]:
+            if caller in spell_targets:
+                caller.msg("You can't cast '%s' on yourself." % spell_to_cast)
+                return
+                
+        # Return if "None" in target list, indicating failed match
+        if None in spell_targets:
+            # No need to give an error message, as 'search' gives one by default.
+            return
+                
+        # Give error message if repeats in target list
+        if len(spell_targets) != len(set(spell_targets)):
+            caller.msg("You can't specify the same target more than once!")
+            return
+        
+        # Finally, we can cast the spell itself. Note that MP is not deducted here!
+        try:
+            spelldata["spellfunc"](caller, spell_to_cast, spell_targets, spelldata["cost"], **kwargs)
+        except Exception:
+            log_trace("Error in callback for spell: %s." % spell_to_cast)
+        """
+        Note: This is a quite long command, since it has to cope with all
+        the different circumstances in which you may or may not be able
+        to cast a spell. None of the spell's effects are handled by the
+        command - all the command does is verify that the player's input
+        is valid for the spell being cast and then call the spell's
+        function.
+        """
+        
+
+class CmdRest(Command):
+    """
+    Recovers damage and restores MP.
+
+    Usage:
+      rest
+
+    Resting recovers your HP and MP to their maximum, but you can
+    only rest if you're not in a fight.
+    """
+
+    key = "rest"
+    help_category = "combat"
+
+    def func(self):
+        "This performs the actual command."
+
+        if is_in_combat(self.caller):  # If you're in combat
+            self.caller.msg("You can't rest while you're in combat.")
+            return
+
+        self.caller.db.hp = self.caller.db.max_hp  # Set current HP to maximum
+        self.caller.db.mp = self.caller.db.max_mp  # Set current MP to maximum
+        self.caller.location.msg_contents("%s rests to recover HP and MP." % self.caller)
+        """
+        You'll probably want to replace this with your own system for recovering HP and MP.
```

Comment:
Since this type of commenting is inconstence with anything else in Evennia, please make this into a comment with `#` or move it together with the regular doc string.